compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -682,7 +710,11 @@ object Objects:
           val tpl = cls.defTree.asInstanceOf[TypeDef].rhs.asInstanceOf[Template]
           extendTrace(cls.defTree) { eval(tpl, ref, cls, cacheResult = true) }
         else
-          extendTrace(ddef) { eval(ddef.rhs, ref, cls, cacheResult = true) }
+          extendTrace(ddef) {
+            Returns.installHandler(ctor)
+            eval(ddef.rhs, ref, cls, cacheResult = true)
+            Returns.popHandler(ctor)
```

Comment:
Minor: Maybe add a comment saying that the return value of constructors is not important.